packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -887,69 +900,167 @@ export class MergeTree {
 	 * @param segments - An array of (not necessarily contiguous) segments with increasing ordinals.
 	 */
 	private slideAckedRemovedSegmentReferences(segments: ISegment[]) {
-		// References are slid in groups to preserve their order.
-		let currentSlideDestination: ISegment | undefined;
-		let currentSlideIsForward: boolean | undefined;
+		function anyLocalRef(
```

Comment:
could you add a note to the doc comment of this function that it does not preserve relative order between references that slide backward and those that slide forward? This is obviously not true for when the references slide to different segments (how could you expect it to be?), but with this implementation it's also not true when both types of references slide to the same place. (i.e. when portions including the start or end of the string are removed). We could make it true with some more effort, but it doesn't seem worth it anyway. I just want to make sure it's documented somewhere :)